### PR TITLE
Add special considerations to scpca-nf user instructions

### LINKED ISSUE #114: scpca-nf user instructions: special considerations for using the spatial workflow
If users would like to analyze spatial transcriptomics data with `scpca-nf`, they will need to have created their own docker container that contains `spaceranger` and updated the parameter for the spaceranger container. We should provide some cautionary guidance that the workflow will not run out of the box for these samples in case users are interested in using that part of the workflow. 

These instructions should be included as a branch on the branch `allyhawkins/outline-scpca-nf-instructions` from https://github.com/AlexsLemonade/scpca-nf/pull/106 so we can merge all the instructions together.

### PR DESCRIPTION
**Pull request by allyhawkins**
Closes #114. 

Here I added in special instructions for using the spatial workflow. I specifically noted that the `.jpg` file must be included in the directory with the FASTQ files and noted the columns that must be added to the `run_metafile`. 

I also included brief instructions on creating a docker image that includes Space Ranger. I wasn't quite sure how much information we wanted to include on actually using Docker itself and creating the image. One thing I thought of was including the link to our Dockerfile in `alsf-scpca` but I wasn't 100% sure if that was okay so didn't include that. Are there things that I should include more detail on? 

In doing this, I also decided to wrap up the special instructions and added a short section at the bottom for using the `repeat_mapping` parameter as was discussed in slack with @jashapiro. Rather than having a section on all of the parameters, I deleted that section and included a short section at the bottom specifically for if users would want to repeat a mapping step they can use the flag at the command line. 

I stacked this on #124 as to minimize merge conflicts.

### CODE REVIEW

**Review by jashapiro — COMMENTED** [review]
I had some smallish updates, mostly wording and organization.

I wanted to add a bit more about docker too, with our example. I did realize though that there could be a limitation with docker where I am not sure if/how nextflow supports private repositories. I don't have any solutions, so I don't even know what to put in the docs...

**Comment by allyhawkins** [comment]
> I wanted to add a bit more about docker too, with our example. I did realize though that there could be a limitation with docker where I am not sure if/how nextflow supports private repositories. I don't have any solutions, so I don't even know what to put in the docs...

I also was really unsure about this... which is why I originally had aired on the side of including less information. I think what you added is fine and then if we find that users are having trouble accessing private repositories we can go from there?

**Comment by allyhawkins** [comment]
I also moved the section on repeat the mapping step and added in the link to our Docker container for Space Ranger like you had suggested. This should be ready for another look.

File: external-data-instructions.md [hunk 1]
```diff
@@ -208,3 +224,5 @@ If bulk libraries are processed, there will be an additional `salmon` folder tha
 
 All files pertaining to a specific library will be nested within a folder labeled with the library ID.
 Additionally, for each run, all files related to that run will be inside a folder labeled with the run ID followed by the type of run (i.e. `rna` or `features` for CITE-seq) and nested within the library ID folder.
+
+If bulk libraries are processed, there will be an additional `salmon` folder that contains the output from running [`salmon`](https://salmon.readthedocs.io/en/latest/file_formats.html#fileformats) on each library processed.
```
Comment (jashapiro) [inline external-data-instructions.md:228]: This snuck back in after moving in the other branch, I think.
```suggestion

```

**Review by jashapiro — APPROVED** [review]
LGTM, less a duplicate line noted below

**Comment by jashapiro** [comment]
Make sure you delete the `allyhawkins/outline-scpca-nf-instructions` branch or manually retarget this before merging!

**Pull request merged**